src/NUnitFramework/framework/Constraints/ThrowsNothingConstraint.cs

```diff
@@ -50,23 +50,9 @@ public override string Description
         /// <returns>True if no exception is thrown, otherwise false</returns>
         public override ConstraintResult ApplyTo<TActual>(TActual actual)
         {
-            //TestDelegate code = actual as TestDelegate;
-            //if (code == null)
-            //    throw new ArgumentException("The actual value must be a TestDelegate", "actual");
-
-            //caughtException = null;
-
-            //try
-            //{
-            //    code();
-            //}
-            //catch (Exception ex)
-            //{
-            //    caughtException = ex;
-            //}
-
-            caughtException = ThrowsConstraint.ExceptionInterceptor.Intercept(actual);
-
+            caughtException = actual as Exception;
+            if (caughtException == null && actual != null)
+                caughtException = ThrowsConstraint.ExceptionInterceptor.Intercept(actual);
```

Comment:
Here's the change mentioned above. I'm not very satisfied with it, but I didn't see a better way to get the correct description ("No Exception to be thrown") without repeating it.